components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/handler/request/PostAuthnMissingChallengeQuestionsHandler.java

```diff
@@ -0,0 +1,420 @@
+    private boolean isChallengeQuestionRequested(AuthenticationContext authenticationContext) {
+
+        return authenticationContext.getParameter(CHALLENGE_QUESTIONS_REQUESTED) != null;
+    }
+
+    /**
+     * Set the Challenge Question Requested parameter in the authenticated context.
+     *
+     * @param authenticationContext Authentication Context.
+     */
+    @SuppressWarnings("unchecked")
+    private void setChallengeQuestionRequestedState(AuthenticationContext authenticationContext) {
+
+        authenticationContext.addParameter(CHALLENGE_QUESTIONS_REQUESTED, true);
+    }
+
+    @Override
+    public String getName() {
+        return "PostAuthnMissingChallengeQuestionsHandler";
+    }
+
+    /**
+     * Returns the property related to the key from the Resident IDP properties.
+     *
+     * @param tenantDomain Tenant Domain.
+     * @param key          Name of the resident IDP property to find in the Resident IDP
+     * @return String value of the requested property.
+     */
+    private String getResidentIdpProperty(String tenantDomain, String key) {
+        IdentityProvider residentIdp;
+
+        try {
+            residentIdp = IdentityProviderManager.getInstance().getResidentIdP(tenantDomain);
+            IdentityProviderProperty[] idpProps = residentIdp.getIdpProperties();
+            for (IdentityProviderProperty property : idpProps) {
+                if (StringUtils.equals(property.getName(), key)) {
+                    return property.getValue();
+                }
+            }
+
+            return null;
+        } catch (IdentityProviderManagementException e) {
+            log.error("Resident IdP value not found. Error while retrieving resident IdP property " +
+                    "for force challenge question ", e);
+            return null;
+        }
+    }
+
+    /**
+     * Returns whether the user has already provided the challenge questions.
+     *
+     * @param user Authenticated User.
+     * @return boolean value indicating whether the user has already provided challenge questions.
+     */
+    private boolean isChallengeQuestionsProvided(AuthenticatedUser user) {
+
+        try {
+            String userName = user.getUserName();
+            int tenantId = Utils.getTenantId(MultitenantUtils.getTenantDomain(userName));
+            UserStoreManager userStoreManager =
+                    IdentityRecoveryServiceDataHolder.getInstance().getRealmService()
+                            .getTenantUserRealm(tenantId)
+                            .getUserStoreManager();
+            Map<String, String> claimsMap = userStoreManager
+                    .getUserClaimValues(userName, new String[]{IdentityRecoveryConstants.CHALLENGE_QUESTION_URI},
+                            UserCoreConstants.DEFAULT_PROFILE);
+            String claimValue = claimsMap.get(IdentityRecoveryConstants.CHALLENGE_QUESTION_URI);
+            return claimValue != null;
+
+        } catch (IdentityException e) {
```

Comment:
Shall we catch Exception in a single line and provide a generic error as below.
catch (IdentityException | UserStoreException e)